### ISSUE TITLE
Hex Encoding

### ISSUE THREAD

**Issue by littletarzan** [hashgraph/hedera-sdk-js#939]
### Description

Also found a potential issue with sdk version 2.9.0-beta.1:

```
const factoryContractId = ContractId.fromString("0.0.20892");
```

when I console.log the contract id, the 2 is being cut off:

`The factory contract ID is 0.0.2089`

using 2.7.0, the entire contract id prints out correctly

### Steps to reproduce

Step through fromString() function call and navigate to line 66 shown here:

https://github.com/hashgraph/hedera-sdk-js/commit/062c306844653825dc9b1398cf8bee23a6daf479#diff-0a3ac5d39220a4638593ab0130611ebc1310e84a11ebd4720240ac0b21cf5b93R64-R66

### Additional context

Please see the result of us tracking down the bug:

`hex.decode(result.numOrHex)`

numOrHex is equal to "20892", which is base10.  It's passing that value into the hex.js encode function, which is calling:

`Buffer.from('20892', 'hex')`

but 20892 is base10 not base16

this demonstrates the issue:
```
const hexdata = Buffer.from('20892', 'hex');
console.log("HEX "+Buffer.from(hexdata).toString('hex'));
> HEX 2089
```
```
const decdata = Buffer.from('20892')
console.log("DEC "+Buffer.from(decdata).toString());
> DEC 20892
```

### Hedera network

previewnet

### Version

v2.9.0-beta.1

### Operating system

macOS

**Comment by danielakhterov** [comment]
Good catch; it seems this mistake was using `Number.isSafeInteger()` on a string which always results in `false`. Will get this fixed before Monday's release.

**Comment by danielakhterov** [comment]
Fixed: 764e68a541e3bd03341fab3888d53464ecee2d2b
